### ISSUE TITLE
500 error with ZIP on Windows installer

### ISSUE THREAD

**Issue by radinamatic** [learningequality/kolibri#1094]
## Summary

Server 500 error on exercises reading ZIP content.

## System information

*Please specify the Kolibri version you were using and your operating system!*

 - Version: 0.2.0.dev3
 - Operating system: Windows 7/8.1/10
 - Browser: all

### Chrome on Win 8.1 
![exercise2](https://cloud.githubusercontent.com/assets/1457929/24067537/809d30b4-0b3b-11e7-868e-1d309b82cb5d.png)

### MSEdge on Win 10

![virtualbox_msedge - win10_preview_17_03_2017_13_40_09](https://cloud.githubusercontent.com/assets/1457929/24067600/1ff63976-0b3c-11e7-9e5e-09f2bb4007aa.png)

![virtualbox_msedge - win10_preview_17_03_2017_17_45_04](https://cloud.githubusercontent.com/assets/1457929/24067621/639218c6-0b3c-11e7-8d49-9b07e6271b45.png)

Error after deleting this condition and restarting the server:
https://github.com/learningequality/kolibri/blob/develop/kolibri/content/views.py#L65

![virtualbox_msedge - win10_preview_17_03_2017_17_46_02](https://cloud.githubusercontent.com/assets/1457929/24067636/bf96dff8-0b3c-11e7-8ca1-84d9d3717632.png)

### Chrome on Win 10 

![virtualbox_msedge - win10_preview_17_03_2017_17_43_17](https://cloud.githubusercontent.com/assets/1457929/24067640/dcdc2f32-0b3c-11e7-8ed4-74688a93ba9c.png)

**Comment by radinamatic** [comment]
Panwar in Uganda UNICEF is still having this issue on 0.3.0 Windows installer:

![error 500 screenshot](https://cloud.githubusercontent.com/assets/1457929/24654429/dd295720-1939-11e7-94ae-bc0a4e07f660.png)


```
INFO 2017-03-08 19:17:43,570 cli Kolibri running for the first time.
INFO 2017-03-08 19:17:43,570 cli We don't yet use pre-migrated database seeds, so you're going to have to wait a bit while we create a blank database...


INFO 2017-03-08 19:17:51,289 cli Automatically enabling applications.
INFO 2017-03-08 19:24:28,062 importchannel Copying in data for channel id 7db80873586841e4a1c249b2bb8ba62d
INFO 2017-03-08 19:28:03,714 importchannel Copying in data for channel id 17e25cd51c1842dd87755dcd7cd515a4
INFO 2017-03-31 17:32:14,772 importchannel Downloading data for channel id 1ceff53605e55bef987d88e0908658c5
INFO 2017-03-31 17:34:11,619 importchannel Downloading data for channel id 1d8f6d84618153c18c695d85074952a7
ERROR 2017-03-31 18:12:07,222 transfer Error reading download stream: HTTPSConnectionPool(host='contentworkshop.learningequality.org', port=443): Read timed out.
ERROR 2017-03-31 19:34:24,466 transfer Error reading download stream: HTTPSConnectionPool(host='contentworkshop.learningequality.org', port=443): Read timed out.
INFO 2017-04-01 10:21:03,297 importchannel Downloading data for channel id 1d8f6d84618153c18c695d85074952a7
ERROR 2017-04-03 12:01:05,297 base Internal Server Error: /zipcontent/904f319eace9da4964b1474d88bb8ab4.zip/
Traceback (most recent call last):
  File "c:\python34\lib\site-packages\kolibri\dist\django\core\handlers\base.py", line 149, in get_response
    response = self.process_exception_by_middleware(e, request)
  File "c:\python34\lib\site-packages\kolibri\dist\django\core\handlers\base.py", line 147, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "c:\python34\lib\site-packages\kolibri\dist\django\views\generic\base.py", line 68, in view
    return self.dispatch(request, *args, **kwargs)
  File "c:\python34\lib\site-packages\kolibri\dist\django\views\generic\base.py", line 88, in dispatch
    return handler(request, *args, **kwargs)
  File "c:\python34\lib\site-packages\kolibri\content\views.py", line 66, in get
    response["Last-Modified"] = http_date(float(datetime.datetime(*info.date_time).strftime("%s")))
ValueError: Invalid format string
ERROR 2017-04-03 12:01:17,618 base Internal Server Error: /zipcontent/d83b48a86f090e25c193efceaecb91e4.perseus/exercise.json
Traceback (most recent call last):
  File "c:\python34\lib\site-packages\kolibri\dist\django\core\handlers\base.py", line 149, in get_response
    response = self.process_exception_by_middleware(e, request)
  File "c:\python34\lib\site-packages\kolibri\dist\django\core\handlers\base.py", line 147, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "c:\python34\lib\site-packages\kolibri\dist\django\views\generic\base.py", line 68, in view
    return self.dispatch(request, *args, **kwargs)
  File "c:\python34\lib\site-packages\kolibri\dist\django\views\generic\base.py", line 88, in dispatch
    return handler(request, *args, **kwargs)
  File "c:\python34\lib\site-packages\kolibri\content\views.py", line 66, in get
    response["Last-Modified"] = http_date(float(datetime.datetime(*info.date_time).strftime("%s")))
ValueError: Invalid format string
ERROR 2017-04-03 12:01:37,907 base Internal Server Error: /zipcontent/d83b48a86f090e25c193efceaecb91e4.perseus/exercise.json
Traceback (most recent call last):
  File "c:\python34\lib\site-packages\kolibri\dist\django\core\handlers\base.py", line 149, in get_response
    response = self.process_exception_by_middleware(e, request)
  File "c:\python34\lib\site-packages\kolibri\dist\django\core\handlers\base.py", line 147, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "c:\python34\lib\site-packages\kolibri\dist\django\views\generic\base.py", line 68, in view
    return self.dispatch(request, *args, **kwargs)
  File "c:\python34\lib\site-packages\kolibri\dist\django\views\generic\base.py", line 88, in dispatch
    return handler(request, *args, **kwargs)
  File "c:\python34\lib\site-packages\kolibri\content\views.py", line 66, in get
    response["Last-Modified"] = http_date(float(datetime.datetime(*info.date_time).strftime("%s")))
ValueError: Invalid format string
ERROR 2017-04-03 12:01:42,571 base Internal Server Error: /zipcontent/904f319eace9da4964b1474d88bb8ab4.zip/
Traceback (most recent call last):
  File "c:\python34\lib\site-packages\kolibri\dist\django\core\handlers\base.py", line 149, in get_response
    response = self.process_exception_by_middleware(e, request)
  File "c:\python34\lib\site-packages\kolibri\dist\django\core\handlers\base.py", line 147, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "c:\python34\lib\site-packages\kolibri\dist\django\views\generic\base.py", line 68, in view
    return self.dispatch(request, *args, **kwargs)
  File "c:\python34\lib\site-packages\kolibri\dist\django\views\generic\base.py", line 88, in dispatch
    return handler(request, *args, **kwargs)
  File "c:\python34\lib\site-packages\kolibri\content\views.py", line 66, in get
    response["Last-Modified"] = http_date(float(datetime.datetime(*info.date_time).strftime("%s")))
ValueError: Invalid format string
ERROR 2017-04-03 12:01:50,639 base Internal Server Error: /zipcontent/8318f2b7c98fe38c2c9b5938360dd1ad.zip/
Traceback (most recent call last):
  File "c:\python34\lib\site-packages\kolibri\dist\django\core\handlers\base.py", line 149, in get_response
    response = self.process_exception_by_middleware(e, request)
  File "c:\python34\lib\site-packages\kolibri\dist\django\core\handlers\base.py", line 147, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "c:\python34\lib\site-packages\kolibri\dist\django\views\generic\base.py", line 68, in view
    return self.dispatch(request, *args, **kwargs)
  File "c:\python34\lib\site-packages\kolibri\dist\django\views\generic\base.py", line 88, in dispatch
    return handler(request, *args, **kwargs)
  File "c:\python34\lib\site-packages\kolibri\content\views.py", line 66, in get
    response["Last-Modified"] = http_date(float(datetime.datetime(*info.date_time).strftime("%s")))
ValueError: Invalid format string
ERROR 2017-04-03 12:02:10,403 base Internal Server Error: /zipcontent/0d28957f26cde871ef7e37c367892f3e.perseus/exercise.json
Traceback (most recent call last):
  File "c:\python34\lib\site-packages\kolibri\dist\django\core\handlers\base.py", line 149, in get_response
    response = self.process_exception_by_middleware(e, request)
  File "c:\python34\lib\site-packages\kolibri\dist\django\core\handlers\base.py", line 147, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "c:\python34\lib\site-packages\kolibri\dist\django\views\generic\base.py", line 68, in view
    return self.dispatch(request, *args, **kwargs)
  File "c:\python34\lib\site-packages\kolibri\dist\django\views\generic\base.py", line 88, in dispatch
    return handler(request, *args, **kwargs)
  File "c:\python34\lib\site-packages\kolibri\content\views.py", line 66, in get
    response["Last-Modified"] = http_date(float(datetime.datetime(*info.date_time).strftime("%s")))
ValueError: Invalid format string
ERROR 2017-04-03 12:02:40,572 base Internal Server Error: /zipcontent/23bc3666425541b9edcf6368826eab49.perseus/exercise.json
Traceback (most recent call last):
  File "c:\python34\lib\site-packages\kolibri\dist\django\core\handlers\base.py", line 149, in get_response
    response = self.process_exception_by_middleware(e, request)
  File "c:\python34\lib\site-packages\kolibri\dist\django\core\handlers\base.py", line 147, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "c:\python34\lib\site-packages\kolibri\dist\django\views\generic\base.py", line 68, in view
    return self.dispatch(request, *args, **kwargs)
  File "c:\python34\lib\site-packages\kolibri\dist\django\views\generic\base.py", line 88, in dispatch
    return handler(request, *args, **kwargs)
  File "c:\python34\lib\site-packages\kolibri\content\views.py", line 66, in get
    response["Last-Modified"] = http_date(float(datetime.datetime(*info.date_time).strftime("%s")))
ValueError: Invalid format string
ERROR 2017-04-03 12:02:57,090 base Internal Server Error: /zipcontent/c347a4605fb95d308bbe3b8935f728ff.zip/
Traceback (most recent call last):
  File "c:\python34\lib\site-packages\kolibri\dist\django\core\handlers\base.py", line 149, in get_response
    response = self.process_exception_by_middleware(e, request)
  File "c:\python34\lib\site-packages\kolibri\dist\django\core\handlers\base.py", line 147, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "c:\python34\lib\site-packages\kolibri\dist\django\views\generic\base.py", line 68, in view
    return self.dispatch(request, *args, **kwargs)
  File "c:\python34\lib\site-packages\kolibri\dist\django\views\generic\base.py", line 88, in dispatch
    return handler(request, *args, **kwargs)
  File "c:\python34\lib\site-packages\kolibri\content\views.py", line 66, in get
    response["Last-Modified"] = http_date(float(datetime.datetime(*info.date_time).strftime("%s")))
ValueError: Invalid format string
ERROR 2017-04-03 12:37:57,684 base Internal Server Error: /zipcontent/d6d123cbefaac57900d9f4bbf04dd2f7.zip/
Traceback (most recent call last):
  File "c:\python34\lib\site-packages\kolibri\dist\django\core\handlers\base.py", line 149, in get_response
    response = self.process_exception_by_middleware(e, request)
  File "c:\python34\lib\site-packages\kolibri\dist\django\core\handlers\base.py", line 147, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "c:\python34\lib\site-packages\kolibri\dist\django\views\generic\base.py", line 68, in view
    return self.dispatch(request, *args, **kwargs)
  File "c:\python34\lib\site-packages\kolibri\dist\django\views\generic\base.py", line 88, in dispatch
    return handler(request, *args, **kwargs)
  File "c:\python34\lib\site-packages\kolibri\content\views.py", line 66, in get
    response["Last-Modified"] = http_date(float(datetime.datetime(*info.date_time).strftime("%s")))
ValueError: Invalid format string
ERROR 2017-04-03 16:19:13,699 base Internal Server Error: /zipcontent/7c4ad21757810c714fbabae299e0829e.perseus/exercise.json
Traceback (most recent call last):
  File "c:\python34\lib\site-packages\kolibri\dist\django\core\handlers\base.py", line 149, in get_response
    response = self.process_exception_by_middleware(e, request)
  File "c:\python34\lib\site-packages\kolibri\dist\django\core\handlers\base.py", line 147, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "c:\python34\lib\site-packages\kolibri\dist\django\views\generic\base.py", line 68, in view
    return self.dispatch(request, *args, **kwargs)
  File "c:\python34\lib\site-packages\kolibri\dist\django\views\generic\base.py", line 88, in dispatch
    return handler(request, *args, **kwargs)
  File "c:\python34\lib\site-packages\kolibri\content\views.py", line 66, in get
    response["Last-Modified"] = http_date(float(datetime.datetime(*info.date_time).strftime("%s")))
ValueError: Invalid format string
ERROR 2017-04-03 16:19:17,103 base Internal Server Error: /zipcontent/53e87fa944a30dab92d2bcfb2bb82cbf.zip/
Traceback (most recent call last):
  File "c:\python34\lib\site-packages\kolibri\dist\django\core\handlers\base.py", line 149, in get_response
    response = self.process_exception_by_middleware(e, request)
  File "c:\python34\lib\site-packages\kolibri\dist\django\core\handlers\base.py", line 147, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "c:\python34\lib\site-packages\kolibri\dist\django\views\generic\base.py", line 68, in view
    return self.dispatch(request, *args, **kwargs)
  File "c:\python34\lib\site-packages\kolibri\dist\django\views\generic\base.py", line 88, in dispatch
    return handler(request, *args, **kwargs)
  File "c:\python34\lib\site-packages\kolibri\content\views.py", line 66, in get
    response["Last-Modified"] = http_date(float(datetime.datetime(*info.date_time).strftime("%s")))
ValueError: Invalid format string
```

Trying to find the PR that fixed this issue, and coming up empty...
@jamalex @indirectlylit @jayoshih @ralphiee22

**Comment by indirectlylit** [comment]
`%s` is a [linux-specific](https://linux.die.net/man/3/strftime) formatting option, and is [not supported by Windows](https://msdn.microsoft.com/en-us/library/fe06s4ak.aspx).

The line was removed in develop but not release here: https://github.com/learningequality/kolibri/pull/1102

**Comment by indirectlylit** [comment]
Looks like [we'd decided not to put it in release](https://github.com/learningequality/kolibri/pull/1093#issuecomment-287850857) because we weren't expecting windows user of 0.3.x. Sounds like that's not the case so I'll open a PR and we can get it into 0.3.1.

**Comment by rtibbles** [comment]
Finally completely fixed in #1197.
